woocommerce/woocommerce
# Manually init template registry and controller for non-block themes

### Changes proposed in this Pull Request:

Fixes https://github.com/woocommerce/woocommerce/issues/50270

This PR addresses an issue where `get_query_template` fails to return anything for non-block themes by manually initializing the BlockTemplatesRegistry and BlockTemplatesController

This change is required due to the recent change in how/when both controllers are initialized. See https://github.com/woocommerce/woocommerce/pull/48905
 








### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Navigate to `WooCommerce -> Settings -> Site Visibility` and make sure the site is in `coming soon` mode.

#### Block theme
3. Navigate to `Appearance -> Themes` and activate a block theme such as Twenty Twenty-Four.
4. Open a new incognito session and visit the frontend.
5. Confirm you see the coming soon mode page.

#### Classic theme
6. Navigate to `Appearance -> Themes` and activate a classic theme such as Storefront.
7. Open a new incognito session and visit the frontend.
8. Confirm you see the coming soon mode page.

#### Regression testing
Repeat testing instructions in https://github.com/woocommerce/woocommerce/pull/48905




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Manually initialize block template registry and controller for non-block themes.

</details>
